extensions-contrib/parquet-extensions/src/main/java/io/druid/data/input/parquet/ParquetHadoopInputRowParser.java

```diff
@@ -56,15 +66,46 @@ public ParquetHadoopInputRowParser(
     }
   }
 
+
+  private LogicalType determineTimestampSpecLogicalType(Schema schema, String timestampSpecField)
+  {
+    for (Schema.Field field : schema.getFields()) {
+      if (field.name().equals(timestampSpecField)) {
+        this.timestampSpecLogicalType = field.schema().getLogicalType();
+        return this.timestampSpecLogicalType;
+      }
+    }
+    // The timestamp field does not exist in the provided schema
+    return null;
+  }
+
   /**
    * imitate avro extension {@link AvroStreamInputRowParser#parseGenericRecord(GenericRecord, ParseSpec, List, boolean, boolean)}
    */
   @Override
   public InputRow parse(GenericRecord record)
   {
+    // We don't want to determine the type every time if we already done so
+    LogicalType logicalType = this.timestampSpecLogicalType;
+    if (null == logicalType) {
+      logicalType = determineTimestampSpecLogicalType(record.getSchema(), timestampSpec.getTimestampColumn());
```

Comment:
If it is guaranteed that schema is equal for all processing records, could we not call `determineTimestampSpecLogicalType()` on each record, if calling this method on the first record returned `null`?